utils/delete_resources.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+# SPDX-FileCopyrightText: 2025 SAP edge team
+#
+# SPDX-License-Identifier: Apache-2.0
+set -euo pipefail
+
+# Reusable JSON patch to remove metadata finalizers
+REMOVE_METADATA_FINALIZERS_PATCH='[{"op": "remove", "path": "/metadata/finalizers"}]'
+
+# Reusable JSON patch to remove spec finalizers (namespaces)
+REMOVE_SPEC_FINALIZERS_PATCH='[{"op": "remove", "path": "/spec/finalizers"}]'
+
+# Usage: ./force-delete-project.sh <project_name>
+PROJECT=${1:-}
+
+if [[ -z "$PROJECT" ]]; then
+  echo "Usage: $0 <project_name>"
+  exit 1
+fi
+
+if ! oc get project "$PROJECT" &>/dev/null; then
+  echo "Project '$PROJECT' not found; nothing to do."
+  exit 0
+fi
+
+echo "Fetching project details for '$PROJECT'..."
+PROJECT_JSON=$(oc get project "$PROJECT" -o json)
+
+RESOURCE_LINES=$(echo "$PROJECT_JSON" | jq -r '.status.conditions[]? | select(.message | test("Some resources are remaining:")) | .message' || true)
+FINALIZER_LINES=$(echo "$PROJECT_JSON" | jq -r '.status.conditions[]? | select(.message | test("Some content in the namespace has finalizers remaining:")) | .message' || true)
+ALL_LINES=$(printf "%s\n%s" "$RESOURCE_LINES" "$FINALIZER_LINES")
+
+if [[ -z "$ALL_LINES" ]]; then
+  echo "No dangling resources or finalizers found."
+else
+  echo "Detected remaining resources:"
+  echo "$ALL_LINES"
+fi
+
+# Handle core kinds mentioned by the namespace controller messages
+echo "Force deleting all pods in '$PROJECT'..."
+oc delete pods -n "$PROJECT" --all --force --grace-period=0 || true
+
+if echo "$ALL_LINES" | grep -qE '\bpersistentvolumeclaims\.'; then
+  echo "Removing finalizers from PVCs in '$PROJECT'..."
+  oc get pvc -n "$PROJECT" -o name 2>/dev/null | while read -r PVC; do
+    echo "Patching $PVC to remove finalizers..."
+    oc patch "$PVC" -n "$PROJECT" --type json -p "$REMOVE_METADATA_FINALIZERS_PATCH" || true
+    oc delete "$PVC" -n "$PROJECT" --force --grace-period=0 || true
+  done
+fi
+
+# Clean up custom resources referenced in the status messages (grouped resources)
+RESOURCE_TYPES=$(echo "$ALL_LINES" | grep -oE '([a-zA-Z0-9]+\.)+[a-zA-Z0-9]+' | sort -u)
```

Comment:
To be more cetain of what's left in the cluster, you already have the oc api-resources command below. You don't need to do that according to the project's status logs which might not as accurate as oc api-resources command. Therefore you can remove this part.